### PR TITLE
VBLOCKS-2560: Migration guide for beta.4 release

### PR DESCRIPTION
**Pull request by afalls-twilio**
## Submission Checklist

 - [x] Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code
 - [ ] Tested code changes and observed expected behavior in the example app
 - [ ] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

Added a migration guide for the up coming beta.4 release.

## Breakdown

- Added migration guide for beta.4 release
- Updated changelog

## Validation

n/a

## Additional Notes

None

### CODE REVIEW

File: docs/migration-guide-beta.4.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+* Application & Main Activity Lifecycle/Event Handling 
+
+For an example on how this is done, please look at the [reference application implementation](https://github.com/twilio/twilio-voice-react-native-app/tree/main)
+or refer to the [pull request](https://github.com/twilio/twilio-voice-react-native-app/pull/129) 
+demonstrating these changes.
+
+#### AndroidManifest.xml Changes
+In previous releases, users of the SDK had to manually update their application's AndroidManifest.xml 
+to reflect changes in the SDK. These changes included, additional permissions or when 
+Services, BroadcastReceivers, or additional Activities were added to the SDK. this is no longer the
```
Comment (mhuynh5757) [inline docs/migration-guide-beta.4.md:18]:
```suggestion
Services, BroadcastReceivers, or additional Activities were added to the SDK. This is no longer the 
```

File: docs/migration-guide-beta.4.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+in the reference application.
+
+#### Encapsulation of Application & Activity Lifecycle Management
+Now applications using the SDK just need to construct two objects, `VoiceActivityProxy` for the 
+application's main activity and `VoiceApplicationProxy` for the application's 
+`android.app.Application` events.
+
+###### How to wire up VoiceActivityProxy to your Main Activity
+Add the `VoiceActivityProxy` as private member to your main activity class and construct it in
+either the constructor (not in the `OnCreate(...)` method). Then, the following activity methods
```
Comment (mhuynh5757) [inline docs/migration-guide-beta.4.md:43]:
```suggestion
either the constructor (not in the `onCreate(...)` method). Then, the following activity methods
```

File: docs/migration-guide-beta.4.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+need to be overridden, `OnCreate(...)`, `OnDestroy(...)` and `onNewIntent(...)`. In the
```
Comment (mhuynh5757) [inline docs/migration-guide-beta.4.md:44]:
```suggestion
need to be overridden, `onCreate(...)`, `onDestroy(...)` and `onNewIntent(...)`. In the 
```

File: docs/migration-guide-beta.4.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+either the constructor (not in the `OnCreate(...)` method). Then, the following activity methods
+need to be overridden, `OnCreate(...)`, `OnDestroy(...)` and `onNewIntent(...)`. In the 
+implementation of these methods, please call the corresponding matching methods in the 
+`VoiceActivityProxy` member object. For a complete example, please refer to the reference 
+application [here](https://github.com/twilio/twilio-voice-react-native-app/blob/main/app/android/app/src/main/java/com/twiliovoicereactnativereferenceapp/MainActivity.java).
+
+###### How to wire up VoiceApplicationProxy to android.app.Application
+Similar to steps for wiring up the `VoiceActivityProxy`, add the `VoiceApplicationProxy` as a 
+private member to your Application class and construct it in the constructor. Then, the following
+`android.app.Application` methods need to be overridden, `OnCreate(...)` & `OnTerminate(...)`. In
```
Comment (mhuynh5757) [inline docs/migration-guide-beta.4.md:52]:
```suggestion
`android.app.Application` methods need to be overridden, `onCreate(...)` & `onTerminate(...)`. In
```

**Review by mhuynh5757 — CHANGES_REQUESTED** [review]
Are the suggestions correct? I recall the methods being camelCased and not PascalCased.

**Review by charliesantos — APPROVED** [review]

**Review by mhuynh5757 — APPROVED** [review]

File: docs/migration-guide-beta.4.md (hunk 4 above)
Comment (afalls-twilio) [reply docs/migration-guide-beta.4.md]: yeah, you are correct!

**Pull request merged**